app/models/user.rb

```diff
@@ -9,33 +9,48 @@ class User < ApplicationRecord
   has_many :posts
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
+  has_many :sent_requests, class_name: 'Friendship', foreign_key: 'user_id', dependent: :destroy
+  has_many :received_requests, class_name: 'Friendship', foreign_key: 'friend_id', dependent: :destroy
 
-  has_many :friendships
-  has_many :inverse_friendships, class_name: 'Friendship', foreign_key: 'friend_id'
-
-  def friends
-    friends_array = friendships.map { |friendship| friendship.friend if friendship.confirmed }
-    friends_array + inverse_friendships.map { |friendship| friendship.user if friendship.confirmed }
-    friends_array.compact
+  def accepted_friendships
+    sent_requests.where(confirmed: true)
   end
 
-  # Users who have yet to confirme friend requests
-  def pending_friends
-    friendships.map { |friendship| friendship.friend unless friendship.confirmed }.compact
+  # Users who has SENT a friend request and is waiting for confirmation
+  def pending_request
+    sent_requests.where(confirmed: nil)
```

Comment:
- [ ] [OPTIONAL] `User#pending_friends` can be defined as association with SQL query instead of processing objects in application memory (as the database was created to store and query data it is always more efficient than ruby)
```ruby
# Users who needs to confirm friendship
has_many :pending_friendships, -> { where confirmed: false }, class_name: "Friendship", foreign_key: "user_id"
has_many :pending_friends, through: :pending_friendships, source: :friend
```